### PR TITLE
Preparing for flutter_map v6

### PR DESCRIPTION
**Pull request by rorystephenson**
@TesteurManiak I'm trying my own packages against the proposed flutter_map refactor branch and since some of my packages use this one in the examples I have started migrating it as well. I'm not sure how you want to handle deprecations and whether you want to deprecate the `centerOnPoints` method or not, in any case I figured I'd open this PR in case the changes can be useful when v6 is eventually released.

### CODE REVIEW

File: lib/src/animated_map_controller.dart [hunk 1]
```diff
@@ -100,15 +100,15 @@ class AnimatedMapController {
...
+    final effectiveZoom = zoom ?? mapController.camera.zoom;
     final effectiveRotation = rotation ?? this.rotation;
     final latLngTween = LatLngTween(
-      begin: mapController.center,
+      begin: mapController.camera.center,
       end: effectiveDest,
     );
     final zoomTween = Tween<double>(
-      begin: mapController.zoom,
+      begin: mapController.camera.zoom,
```
Comment (TesteurManiak) [inline lib/src/animated_map_controller.dart:111]: Maybe directly create a variable that would hold `mapController.camera` to reduce verbosity:
```dart
final camera = mapController.camera;

final effectiveDest = dest ?? camera.center;
final effectiveZoom = zoom ?? camera.zoom;
```

**Review by TesteurManiak — CHANGES_REQUESTED** [review]
Thanks for the contribution 😄 
I'm okay with the deprecation of the methods `animatedFitBounds` and `centerOnPoints`. Got a minor remark on the coding style but otherwise this PR seems good to me and I'll merge it once the v6 will be released 👍

**Comment by rorystephenson** [comment]
My pleasure, I've made that change so this is ready once v6 is released.

**Review by TesteurManiak — APPROVED** [review]

**Comment by rorystephenson** [comment]
@TesteurManiak I've changed the dependency from by branch to 6.0.0-dev.1. In doing so it changed the example's pubspec.lock flutter version constraint to >= 3.10.0. Not 100% why, I've left it there but feel free to change it.

**Pull request merged**